docker/qgis/Dockerfile

```diff
@@ -0,0 +1,40 @@
+# For finding latest versions of the base image see
+# https://github.com/SwissDataScienceCenter/renkulab-docker
+ARG BASE_IMAGE=renku/renkulab-vnc:latest
+FROM ${BASE_IMAGE}
+
+USER root
+
+# Install qGIS
+RUN apt-get update \
+    && apt-get install -y -q --no-install-recommends \
+        gnupg \
+        software-properties-common \
+        libqt5gui5 \
+    && apt-get autoremove --purge \
+    && apt-get clean \
+    && rm -rf /var/lib/apt/lists/* \
+    && rm -rf /tmp/*
+
+RUN wget -qO - https://qgis.org/downloads/qgis-2021.gpg.key | sudo gpg --no-default-keyring --keyring gnupg-ring:/etc/apt/trusted.gpg.d/qgis-archive.gpg --import \
+    && chmod a+r /etc/apt/trusted.gpg.d/qgis-archive.gpg
+
+RUN add-apt-repository "deb https://qgis.org/ubuntu $(lsb_release -c -s) main"
+
+# Fixes a bug https://github.com/qgis/QGIS/issues/35649 and removes qgis-providers which may be a problem
+# RUN apt-get remove --auto-remove libproj15 libproj-dev qgis-providers
+
+# Fixes downstream bug with libQt5Core.so.5
+RUN strip --remove-section=.note.ABI-tag /usr/lib/x86_64-linux-gnu/libQt5Core.so.5
```

Comment:
can you move this to above line 9 and just do `apt-get update` + `install` once?

Reply:
I don't think so, since `add-apt-repository` requires `software-properties-common` so you have to do it in two stages. I just tried what you suggested and I get
```
Step 5/11 : RUN add-apt-repository "deb https://qgis.org/ubuntu $(lsb_release -c -s) main"
 ---> Running in b4fd527615ba
/bin/bash: add-apt-repository: command not found
The command '/bin/bash -o pipefail -c add-apt-repository "deb https://qgis.org/ubuntu $(lsb_release -c -s) main"' returned a non-zero code: 127
```

Reply:
I see - but you can still do it as part of the same `RUN` statement above and clean up all the apt stuff afterwards I think?

Reply:
Something like this?
```
RUN apt-get update \
    && apt-get install -y -q --no-install-recommends \
        gnupg \
        software-properties-common \
        libqt5gui5 \
    && wget -qO - https://qgis.org/downloads/qgis-2021.gpg.key | gpg --no-default-keyring --keyring gnupg-ring:/etc/apt/trusted.gpg.d/qgis-archive.gpg --import \
    && chmod a+r /etc/apt/trusted.gpg.d/qgis-archive.gpg \\
    && add-apt-repository "deb https://qgis.org/ubuntu $(lsb_release -c -s) main" \
    && apt-update \
    && apt-get install -y -q --no-install-recommends \
        qgis \
    && apt-get autoremove --purge \
    && apt-get clean \
    && rm -rf /var/lib/apt/lists/* \
    && rm -rf /tmp/*
```

Reply:
yep